### ISSUE TITLE
max_grad_norm schedule

### ISSUE THREAD

**Issue by ffuuugor** [pytorch/opacus#375]
`max_grad_norm` is functionally very similar to `noise_multplier` - it's also stored in the optimizer field as a scalar and never updated.

We have convenient scheduling option for optimizer, but not for `max_grad_norm`.
I can't point to any particular paper proposing this, but some (e.g. https://arxiv.org/pdf/2108.01624.pdf) show that over time singal-to-noise ratio can decline due to natural decline in gradient norm (expected in many training scenarions)
